### PR TITLE
*: sort all the imports

### PR DESCRIPTION
**Pull request by benesch**
I'm submitting this PR as a starting point for discussion. I've added a new flag to crlfmt, -groupimports, that performs import sorting automatically. That way adventurous folks, like myself and @jordanlewis, can opt-in for a bit and see how many spurious diffs are generated in the normal course of development. 

As I mentioned on Slack, think that teaching crlfmt to do this automatically will be a net positive. crlfmt is already picky enough about function signatures that I think most everyone has their editors configured to format automatically on save, or at least have a workflow that involves running crlfmt before pushing to TeamCity. If crlfmt is good enough at sorting imports, no one will ever think about import sorting again. That's not the case in the current "try not to worry about import sorting" state of affairs. I don't nit import sorting on code reviews, but I still end up wasting time sorting imports if I add/remove an import to a file with a mis-sorted list of imports—I just can't help it.

---

Pick up a new version of crlfmt which consistently sorts all of our
imports into two groups: standard library and non-standard library
imports.

This import sorting is compatible with both gofmt and goimports; that
is, upon seeing this import sorting, neither gofmt nor goimports will
feel compelled to change it. In fact, recent versions of goimports will
attempt to create the same grouping, though goimports is less smart
about combining multiple import blocks together.

Note that the crlfmt bump also picks up a fix for a minor bug in how
shorthand function arguments are formatted.

Release note: None

### CODE REVIEW

File: pkg/sql/pgwire/pgerror/main_test.go [hunk 1]
```diff
@@ -13,10 +13,8 @@
...
-	// --vmodule flag and friends. As of 03/2017, no test in this package needs
-	// logging, but we want the test binary to accept the flags for uniformity
-	// with the other tests.
-	_ "github.com/cockroachdb/cockroach/pkg/util/log"
-)
+import // Import the logging package so that tests in this package can accept the
+// --vmodule flag and friends. As of 03/2017, no test in this package needs
+// logging, but we want the test binary to accept the flags for uniformity
+// with the other tests.
+_ "github.com/cockroachdb/cockroach/pkg/util/log"
```
Comment (benesch) [inline pkg/sql/pgwire/pgerror/main_test.go]: Whoops. I'll have to fix this.

**Comment by bdarnell** [comment]
> Consistently sort all of our imports into three groups. The first group
contains stdlib imports. The second group contains third-party imports.
The third group contains imports cockroachdb imports.

I understand that having three groups with some imports in the "wrong" group is an eyesore. But I'd rather solve this by forbidding the third group than mandating it. Why keep cockroachdb imports separate?

**Comment by benesch** [comment]
> Why keep cockroachdb imports separate?

Sure, that works for me.

**Review by knz — APPROVED** [review]

**Review by jordanlewis — APPROVED** [review]
LGTM (once you re-run the program to sort the imports)

**Comment by benesch** [comment]
Alright, this is rebased and looking good. Taking the opportunity to get this merged, since most folks are busy with the Day of Roach celebrations.

bors r=jordanlewis,knz

**Comment by jordanlewis** [comment]
To a glorious import sorting free future! 🍻

**Pull request merged**